### PR TITLE
React native notifications

### PR DESCRIPTION
**Pull request by raymondjacobson**
Corresponding client PR: https://github.com/AudiusProject/audius-client/pull/232

Currently live on Bounce (will update CI commands to disallow bounce from branches)

TODOs:
https://www.notion.so/audiusproject/TODO-list-f951b278485e4bf9bbfc244172957df9

### CODE REVIEW

File: src/components/notifications/Notifications.tsx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+const mapStateToProps = (state: AppState) => ({
+  isOpen: getIsOpen(state)
+})
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  open: () => dispatch(notificationsActions.open()),
+  close: () => dispatch(notificationsActions.close()),
+  markAsViewed: () => dispatch(notificationsActions.markAsViewed())
+})
+
+export default connect(mapStateToProps, mapDispatchToProps)(Notifications)
```
Comment (jowlee) [inline src/components/notifications/Notifications.tsx:286]: Could be using `useSelector`, `useDispatch` if you wanted to 🙃  - no need to change.

File: src/components/notifications/content/TwitterShare.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  const text = `My track ${notification.entity.title} is trending ${
+    notification.rank
+  }${getRankSuffix(
+    notification.rank
+  )} on @AudiusProject! #AudiusTrending #Audius`
+  return { link, text }
+}
+
+const getTwitterHandleByUserHandle = async (userHandle: string) => {
+  return 'blah'
```
Comment (jowlee) [inline src/components/notifications/content/TwitterShare.tsx:68]: blah

**Review by jowlee — APPROVED** [review]
Looks great! Thanks for doing this, it was a huge overhaul - moving over the types too👌 !

Sometimes the text is all below the image and sometime it is inline. ie. Artist Update in the screenshots below. I think it might be fine, just wanted to flag.
Also, in the screenshot for new Remix, the image and text don't appear to be aligned

 
IMG_B0EF091B5449-1

IMG_0288

File: .env.dev [hunk 3]
```diff
@@ -2,4 +2,6 @@ FCM_SENDER_ID=123456789012
 SEGMENT_IOS_WRITE_KEY=abcdef
 SEGMENT_ANDROID_WRITE_KEY=abcdef
 
-URL_OVERRIDE=http://localhost:3000
\ No newline at end of file
+URL_OVERRIDE=http://ray-macbookpro:3001
```
[the remark this answers is not in the document]
Comment (raymondjacobson) [reply .env.dev:5]: it does, sorry this was just to get my phone to connect. will revert before merge

File: src/components/notifications/List.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            style={styles.footer}
+          >
+            <LottieView
+              source={require('../../assets/animations/loadingSpinner.json')}
+              autoPlay
+              loop
+              colorFilters={[
+                {
+                  keypath: 'Shape Layer 1',
+                  color: '#C2C0CC'
```
[the remark this answers is not in the document]
Comment (raymondjacobson) [reply src/components/notifications/List.tsx:118]: ty

**Pull request merged**